### DISCUSSION TITLE
Commandbox server start after CFWheels install starting wrong webroot

### DISCUSSION THREAD

**Discussion by cfcoder** [wheels-dev/wheels#1309] [General]
I've been using commandbox for other web application CF server instances, till now. I've created a new directory, start commandbox and install the cfwheels-cli first, then generate a wheels app, then from the webroot directory, did a server start command.  It comes up servering a completely different web app, one of the former applications in a completely different directory tree structure. How is that happening? Is commandbox somehow configured for a root tree and starting a previous instance of a web app in that tree? 

Very puzzling!

**Comment by cfcoder** [comment]
I removed all the files that generate created in wwwroot. Then regenerated the CFWheels app and called it myApp. This created the CFWheels directory myApp inside of my wwwroot directory where my pwd was. I copied all the contents of  myApp dir and moved it up one level into wwwroot. Quit out of Commandbox and deleted myApp dir (now empty). Then restarted Commandbox and from wwwroot, gave the start command and up came the CFWheels default page!!

I have no idea what was going on. So problem appears to be resolved but no clue why that bizarre behavior was happening.

**Comment by dbelanger** [comment]
Hey Bruce, I don't believe that's Wheels, that's probably CommandBox.  Basically, the commandbox generated server name would use your app name.  If no app name is present,  it uses the root directory directory.   So if you already had started another app with the same app name or directory,  voila, it starts that app in the other directory every time.

That's actually handy if you're in another project and you need to start a depdency (like an API for a front end)
________________________________

**Reply by cfcoder** [reply]
I'm not sure what you mean by commandbox generated server name. Since I'm not remapping the host it generally (in my case) starts the CF server with an IP:port URL. But for the wheels generated app I had specified wwwroot as my app name (wheels generate app wwwroot) as I recall. But that's the name of the webroot for all my instances. Perhaps that confused it?

**Comment by dbelanger** [comment]
That's my guess.  To see a list of commandbox generated servers, just type "server list" - I'm sure you had another server in there with the same name.  That's usually the only reason a server starts "in another directory"




David Paul Bélanger, BCom, EMBA

Web Developer

Belatech Innovations Inc.<https://www.belatech.com/>



Canada: +1 (519) 380-5840<tel:+15193805840>

USA: +1 (954) 603-7476<tel:+19546037476>

Latin America: +54 (911) 5000-2173<tel:+5491150002173>

________________________________

**Reply by cfcoder** [reply]
Thanks!

**Reply by bpamiri** [reply]
Another handy command to know is “server forget <server name>” That allows you to delete a previously created server.
